### PR TITLE
Added `set_minimized` and `set_position` to `Window`

### PR DESCRIPTION
**Pull request by Toniman20**

### CODE REVIEW

**Comment by bjorn3** [comment]
Wayland itself doesn't have a way to set a global position of a window. This is because it was made with the idea that a screen could have arbitrary shapes. Some compositors do provide an extension to set the global position, but this is not something you can depend on. It also doesn't make much sense for tiling compositors I think.

**Review by alice-i-cecile — APPROVED** [review]
This is useful functionality, and the code quality and comments look good.

A shame about the limited functionality on some platforms, but I think that supporting this on the platforms that do allow it is a clear improvement.

File: crates/bevy_window/src/window.rs [hunk 1]
```diff
@@ -199,12 +210,51 @@ impl Window {
         self.physical_height
     }
 
+    /// The window's client position on the x axis in physical pixels.
+    #[inline]
+    pub fn x(&self) -> i32 {
```
Comment (cart) [inline crates/bevy_window/src/window.rs:215]: This seems like as good a time as any to start adopting glam's new "int vec" types. I like the idea of having a single `position() -> IVec2` here (and all other "position" apis added in this pr).

The plan is to slowly start migrating to int vecs where appropriate (ex: individual x/y properties like this and "float vecs" that should be "int vecs")

Comment (Toniman20) [reply crates/bevy_window/src/window.rs]: Done.

**Comment by cart** [comment]
Looks good to me. I just pushed a change that sets the position immediately on construction. This prevents the value from being None on the first schedule run.

**Comment by cart** [comment]
I still included Option because some platforms don't support window position.

Although actually we could just treat that as 0,0 and call it a day. That would make the api easier to consume. Feel free to follow up if you're interested :smile:

**Comment by DJMcNab** [comment]
@Toniman20 please respond in #2373 for the relicense to MIT/Apache 2.0. Thanks!

**Pull request merged**